src/patterns/ec2-app.ts

```diff
@@ -9,17 +11,43 @@ import type { Gu5xxPercentageMonitoringProps, NoMonitoring } from "../constructs
 import { Gu5xxPercentageAlarm } from "../constructs/cloudwatch";
 import type { GuStack } from "../constructs/core";
 import { AppIdentity } from "../constructs/core/identity";
-import { GuVpc, SubnetType } from "../constructs/ec2";
+import { GuSecurityGroup, GuVpc, SubnetType } from "../constructs/ec2";
 import { GuGetPrivateConfigPolicy, GuInstanceRole } from "../constructs/iam";
 import {
   GuApplicationLoadBalancer,
   GuApplicationTargetGroup,
   GuHttpsApplicationListener,
 } from "../constructs/loadbalancing";
 
+const PUBLIC = "PUBLIC" as const;
+const RESTRICTED = "RESTRICTED" as const;
```

Comment:
Could we make this an enum or a const with multiple properties to enable the same method of interaction as [`Stage`](https://github.com/guardian/cdk/blob/7336acc94f541831c45d64468b78a53db52a1e8e/src/constants/stage.ts) or [`RegexPattern`](https://github.com/guardian/cdk/blob/7336acc94f541831c45d64468b78a53db52a1e8e/src/constants/regex-pattern.ts)? That is, prefer `AccessScope.PUBLIC` over `"PUBLIC"`.

Reply:
Yep, think that's sensible.